pkg/controller/prometheus.go

```diff
@@ -224,27 +240,20 @@ func (c *Controller) deletePrometheusAddlConfig(ctx context.Context, tenant *min
 		return err
 	}
 
-	var scrapeConfigs []configmaps.ScrapeConfig
+	var scrapeConfigs, exceptedScrapeConfigs []configmaps.ScrapeConfig
 	err = yaml.Unmarshal(secret.Data[miniov2.PrometheusAddlScrapeConfigKey], &scrapeConfigs)
 	if err != nil {
 		return err
 	}
-	// Check if the scrape config is present
-	hasScrapeConfig := false
-	scIndex := -1
-	for i, sc := range scrapeConfigs {
-		if sc.JobName == tenant.PrometheusOperatorAddlConfigJobName() {
-			hasScrapeConfig = true
-			scIndex = i
-			break
+	for _, sc := range scrapeConfigs {
+		if !strings.HasPrefix(sc.JobName, tenant.PrometheusOperatorAddlConfigJobName()) {
```

Comment:
Is this `Addl` spelling intentional, because I see it in more places. If not, then please rename to `Add`. Also `PrometheusAddlScrapeConfigKey`, `checkAndCreatePrometheusAddlConfig`, `deletePrometheusAddlConfig`, ...

Reply:
@ramondeklein @jiuker it is a short form of `Additional`

Reply:
Ah, thanks.